From 032c2f205f72400334254d8390a40fead7d7b3f4 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Wed, 7 Apr 2010 19:46:27 +0800
Subject: [PATCH 05/22] use common rules

 - LF, SP are all defined in RFC 5234 - don't define them again.
 - use HTAB instead of HT - HTAB is another common rule.
 - use obj-id, zero-id, refname - they are defined in protocol-common.txt.
---
 Documentation/technical/http-protocol.txt | 36 +++++++++++--------------------
 1 file changed, 12 insertions(+), 24 deletions(-)

diff --git a/Documentation/technical/http-protocol.txt b/Documentation/technical/http-protocol.txt
index b83dc32..30a1d91 100644
--- a/Documentation/technical/http-protocol.txt
+++ b/Documentation/technical/http-protocol.txt
@@ -190,14 +190,9 @@ the default ref named 'HEAD'.
   info_refs   =  *( ref_record )
   ref_record  =  any_ref | peeled_ref
 
-  any_ref     =  id HT name LF
-  peeled_ref  =  id HT name LF
-                 id HT name "^{}" LF
-  id          =  40*HEX
-
-  HEX         =  "0".."9" | "a".."f"
-  LF          =  <US-ASCII LF, linefeed (10)>
-  HT          =  <US-ASCII HT, horizontal-tab (9)>
+  any_ref     =  obj-id HTAB refname LF
+  peeled_ref  =  obj-id HTAB refname LF
+                 obj-id HTAB refname "^{}" LF
 
 Smart Clients
 ~~~~~~~~~~~~~
@@ -286,24 +281,17 @@ declarations behind a NUL on the first ref.
                      "0000"
    ref_list       =  empty_list | populated_list
 
-   empty_list     =  PKT-LINE(id SP "capabilities^{}" NUL cap_list LF)
+   empty_list     =  PKT-LINE(obj-id SP "capabilities^{}" NUL cap_list LF)
 
-   non_empty_list =  PKT-LINE(id SP name NUL cap_list LF)
+   non_empty_list =  PKT-LINE(obj-id SP name NUL cap_list LF)
                      *ref_record
 
    cap_list       =  *(SP capability) SP
    ref_record     =  any_ref | peeled_ref
 
-   any_ref        =  PKT-LINE(id SP name LF)
-   peeled_ref     =  PKT-LINE(id SP name LF)
-                     PKT-LINE(id SP name "^{}" LF
-   id             =  40*HEX
-
-   HEX            =  "0".."9" | "a".."f"
-   NL             =  <US-ASCII NUL, null (0)>
-   LF             =  <US-ASCII LF,  linefeed (10)>
-   SP             =  <US-ASCII SP,  horizontal-tab (9)>
-
+   any_ref        =  PKT-LINE(obj-id SP name LF)
+   peeled_ref     =  PKT-LINE(obj-id SP name LF)
+                     PKT-LINE(obj-id SP name "^{}" LF
 
 Smart Service git-upload-pack
 ------------------------------
@@ -350,8 +338,8 @@ appear in the response obtained through ref discovery.
    have_list         =  *PKT-LINE("have" SP id LF)
 
    command           =  create | delete | update
-   create            =  40*"0" SP new_id SP name
-   delete            =  old_id SP 40*"0" SP name
+   create            =  zero-id SP new_id SP name
+   delete            =  old_id SP zero-id SP name
    update            =  old_id SP new_id SP name
 
 TODO: Document this further.
@@ -527,8 +515,8 @@ the id obtained through ref discovery as old_id.
   cap_list        =  *(SP capability) SP
 
   command         =  create | delete | update
-  create          =  40*"0" SP new_id SP name
-  delete          =  old_id SP 40*"0" SP name
+  create          =  zero-id SP new_id SP name
+  delete          =  old_id SP zero-id SP name
   update          =  old_id SP new_id SP name
 
 TODO: Document this further.
-- 
1.8.4.rc4.527.g303b16c

